source/common/common/utility.h

```diff
@@ -110,6 +111,27 @@ class RealTimeSource : public TimeSource {
   MonotonicTime monotonicTime() override { return std::chrono::steady_clock::now(); }
 };
 
+/**
+ * Class used for creating non-memory allocating std::ostream.
+ */
+class FixedSizeStreamBuffer : public std::streambuf {
+public:
+  FixedSizeStreamBuffer(char* base, size_t size);
+};
+
+/**
+ * std::ostream class that serializes writes into the provided buffer.
+ */
+class OutputBufferStream : private FixedSizeStreamBuffer, public std::ostream {
+public:
+  OutputBufferStream(char* data, size_t size);
+
+  /**
+   * @return the number of bytes written prior to the "put" pointer into the buffer.
+   */
+  int numValidBytesWrittenToBuffer() const;
```

Comment:
This method name could simply be: bytesWritten()

But given expected use, I would suggest you consider an accessor that returns the written bytes as an absl::string_view.  This would simplify both testing an future use.  You may want to verify that the base pointer of the string view matches the base pointer passed in to the OutputBufferStream constructor.

A possible method name could be "contents"